monai/data/dataset.py

```diff
@@ -422,43 +423,56 @@ def set_data(self, data: Sequence):
 
         """
         super().set_data(data=data)
-        self._read_env = None
+        self._env = None
+        self._read_env = self._fill_cache_start_reader(show_progress=self.progress)
```

Comment:
So it will behave like `CacheDataset` that runs the caching before training instead of runtime?
We use `multi-threads` in `CacheDataset` to accelerate the caching progress, as you said the multi-processing perf is not good, is it possible to support multi-threads like `CacheDataset`?

Thanks.

Reply:
sure, I'll try that soon

Reply:
I looked into this, LMDB is single-writer multi-reader by default https://github.com/jnwatson/py-lmdb/issues/114, so I think the current solution is good for now.